optimum/onnxruntime/modeling_ort.py

```diff
@@ -0,0 +1,681 @@
+        for more information.
+"""
+
+ONNX_INPUTS_DOCSTRING = r"""
+    Args:
+        input_ids (`torch.Tensor` of shape `({0})`):
+            Indices of input sequence tokens in the vocabulary.
+            Indices can be obtained using [`AutoTokenizer`](https://huggingface.co/docs/transformers/autoclass_tutorial#autotokenizer). 
+            See [`PreTrainedTokenizer.encode`](https://huggingface.co/docs/transformers/main_classes/tokenizer#transformers.PreTrainedTokenizerBase.encode) and
+            [`PreTrainedTokenizer.__call__`](https://huggingface.co/docs/transformers/main_classes/tokenizer#transformers.PreTrainedTokenizerBase.__call__) for details.
+            [What are input IDs?](https://huggingface.co/docs/transformers/glossary#input-ids)
+        attention_mask (`torch.Tensor` of shape `({0})`, *optional*):
+            Mask to avoid performing attention on padding token indices. Mask values selected in `[0, 1]`:
+            - 1 for tokens that are **not masked**,
+            - 0 for tokens that are **masked**.
+            [What are attention masks?](https://huggingface.co/docs/transformers/glossary#attention-mask)
+        token_type_ids (`torch.Tensor` of shape `({0})`, *optional*):
+            Segment token indices to indicate first and second portions of the inputs. Indices are selected in `[0, 1]`:
+            - 1 for tokens that are **sentence A**,
+            - 0 for tokens that are **sentence B**.
+            [What are token type IDs?](https://huggingface.co/docs/transformers/glossary#token-type-ids)
+"""
+
+
+@add_start_docstrings(
+    """
+    Base OnnxModel class for implementing models using ONNX Runtime. The OnnxModel implements generic methods for interacting
+    with the Hugging Face Hub as well as exporting vanilla transformers models to ONNX using `transformers.onnx` toolchain.
+    The OnnxModel implements additionally generic methods for optimizing and quantizing Onnx models.
+    """,
+)
+class OnnxModel(OptimizedModel):
+    base_model_prefix = "onnx_model"
+
+    def __init__(self, model=None, config=None, **kwargs):
+        self.model = model
+        self.config = config
+        self.model_save_dir = kwargs.get("model_save_dir", None)
+        self.latest_model_name = kwargs.get("latest_model_name", "model.onnx")
+
+    def forward(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def optimize(
+        self,
+        input_path: Union[str, Path] = None,
+        model_type: str = "bert",
+        output_path: Union[str, Path] = default_cache_path,
+        opt_level: int = 1,
+        **kwargs,
+    ):
+        """
+        optimizes the mode using onnxruntime.tools.transformers.optimize
+        Arguments:
+            input_path (:obj:`str` or :obj:`Path`):
+                Directory from which to load
+            output_path(:obj:`str`):
+                Directory where to save the quantized model should be saved, default to
```

Comment:
Typo : `where to save the quantized model` or `where the quantized model should be saved`